### ISSUE TITLE
Difference between PDF and PNG/SVG with patterns and gradients

### ISSUE THREAD

**Issue by EpicEricEE** [typst/typst#2900]
### Description

_(Left to right: pdf, png, svg / using Edge as pdf viewer / commit hash 9e333e5)_

### Pattern Offset
```typ
#set page(width: auto, height: auto, margin: 4mm, fill: white)

#box(
  width: 3cm, height: 3cm,
  fill: pattern(box(
    width: 1cm, height: 1cm,
    fill: gradient.linear(red, blue, angle: 45deg)
  ))
)
```

![image](https://github.com/typst/typst/assets/7191192/07e22126-0a63-4b4b-b0af-77eff67f9829)

### ~~Gradient Offset~~ (Fixed)
```typ
#set page(width: auto, height: auto, margin: 4mm, fill: white)

#box(
  width: 3cm, height: 3cm,
  fill: gradient.linear(red, blue, angle: 45deg)
)
```

![image](https://github.com/typst/typst/assets/7191192/22f04562-a76a-4a44-b34c-16374165b062)

### Operating system

Web app, Windows

### Typst version

- [X] I am using the latest version of Typst

**Comment by MDLC01** [comment]
If I understood properly, this was fixed by #2927

**Comment by EpicEricEE** [comment]
Only the gradient part was fixed; pattern behavior was not changed by that PR.
